### PR TITLE
libtool: uses from macos m4

### PR DESCRIPTION
**Pull request by moonfruit**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/lib/libtool.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ class Libtool < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "f55d5bcc07a45f599800b2c9fb5818c13be90803355e169cdb0e1ddc621bee5e"
   end
 
-  depends_on "m4"
+  uses_from_macos "m4"
```
Comment (carlocab) [inline Formula/lib/libtool.rb:22]:
```suggestion
  # We use Homebrew `m4` for consistency with `automake` and `autoconf`.
  # https://github.com/Homebrew/homebrew-core/commit/da7c013ea7283936782fe15d33d21fd5cd7a97c6
  depends_on "m4"
```

**Review by carlocab — CHANGES_REQUESTED** [review]
Thanks, but I don't think this change is appropriate. See https://github.com/Homebrew/homebrew-core/commit/da7c013ea7283936782fe15d33d21fd5cd7a97c6

**Comment by fxcoudert** [comment]
Agree with @carlocab, the switch to brew `m4` was a careful decision, unless there is significant evidence of problems with that, we should not reverse.

**Comment by moonfruit** [comment]
Maybe I am not right, but I found that the `m4` `libtool` uses at runtime is not brewed `m4` but macos's system `m4`.

`glibtool` doesn't use `m4` at all.
`glibtoolize` uses system `m4`.

I simple put these to `glibtoolize` before `exit $?`, and it output as `/usr/bin/gm4`. This is macos bundled m4 not brewed m4. 

```diff
4365a4366,4367
> func_require_gnu_m4
> echo $M4
```

So I think libtool never use brewed `m4`. If you want to use brewed `m4` for consistency with `automake` and `autoconf`, you need to do some extra work rather than simply adding a dependency.

**Pull request closed without merging**